src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/IOptionDescriptor.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc.OptionDescriptors
+{
+    /// <summary>
+    /// Encapsulates information that describes a <typeparamref name="TOption"/> option on <see cref="MvcOptions"/> .
+    /// </summary>
+    /// <typeparam name="TOption">The type of the option.</typeparam>
+    public interface IOptionDescriptor<out TOption>
```

Comment:
The interface is needed for covariance, so that methods in OptionDescriptorExtensions extending `IEnumerable<IOptionDescriptor<object>>` can be called from a sequence of `InputFormatterDescriptor`, for example.